pkg/controller/factory/factory.go

```diff
@@ -119,22 +124,54 @@ func (f *Factory) WithBareInformers(informers ...Informer) *Factory {
 // Pass the queueKeyFn you want to use to transform the informer runtime.Object into string key used by work queue.
 func (f *Factory) WithInformersQueueKeyFunc(queueKeyFn ObjectQueueKeyFunc, informers ...Informer) *Factory {
 	f.informerQueueKeys = append(f.informerQueueKeys, informersWithQueueKey{
-		informers:  informers,
-		queueKeyFn: queueKeyFn,
+		informers: informers,
+		queueKeysFn: func(obj runtime.Object) []string {
+			key := queueKeyFn(obj)
+			return []string{key}
+		},
+	})
+	return f
+}
+
+// WithInformersQueueKeysFunc is used to register event handlers and get the caches synchronized functions.
+// Pass informers you want to use to react to changes on resources. If informer event is observed, then the Sync() function
+// is called.
+// Pass the queueKeysFn you want to use to transform the informer runtime.Object into a string slice of keys used by work queue.
+func (f *Factory) WithInformersQueueKeysFunc(queueKeysFn ObjectQueueKeysFunc, informers ...Informer) *Factory {
+	f.informerQueueKeys = append(f.informerQueueKeys, informersWithQueueKey{
+		informers:   informers,
+		queueKeysFn: queueKeysFn,
 	})
 	return f
 }
 
 // WithFilteredEventsInformersQueueKeyFunc is used to register event handlers and get the caches synchronized functions.
 // Pass informers you want to use to react to changes on resources. If informer event is observed, then the Sync() function
 // is called.
-// Pass the queueKeyFn you want to use to transform the informer runtime.Object into string key used by work queue.
+// Pass the queueKeysFn you want to use to transform the informer runtime.Object into string key used by work queue.
 // Pass filter to filter out events that should not trigger Sync() call.
 func (f *Factory) WithFilteredEventsInformersQueueKeyFunc(queueKeyFn ObjectQueueKeyFunc, filter EventFilterFunc, informers ...Informer) *Factory {
 	f.informerQueueKeys = append(f.informerQueueKeys, informersWithQueueKey{
```

Comment:
s/informersWithQueueKey/informersWithQueueKeys/

Reply:
this is still to use a func that returns single key